Server/src/services/authenticationService.ts

```diff
@@ -0,0 +1,142 @@
+import { ILoginInformation } from '../genericInterfaces/AuthenticationInterfaces';
+import { IJwtParams } from '../genericInterfaces/AuthenticationInterfaces';
+
+import { IResponse } from '../genericInterfaces/ResponsesInterface'
+
+
+/**
+ * This class services authentication or User related requests and handles
+ * interacts with the database for the Accounts table
+ */
+export class AuthenticationService {
+    private requestResponse: IResponse = {} as any;
+
+    constructor() {
+    }
+
+    /**
+     * This method is responsible for handling the logic of registering a new user. Steps followed:
+     * User input email is checked against the database for duplication, upon success, the user password is 
+     * hashed and the user is registered.
+     * @param SignUpInformation - User Sign in Information. ISignUp is an interface with all required params
+     */
+    async processSignUp(SignUpInformation: ISignUpInformation): Promise<IResponse> {
+
+        let email: boolean;
+        try {
+            email = await AuthenticationModel.verifyIfEmailExists(SignUpInformation.email);
+            if (!email) {
+                SignUpInformation.password = await this.hashPassword(SignUpInformation.password)
+                await AuthenticationModel.insertSignUpInformation(SignUpInformation);
+            }
+            else throw new Error();
+
+        } catch (error) {
+            this.requestResponse.status = "Failure";
+            this.requestResponse.statusCode = 400;
+            this.requestResponse.response = "Email is already in the System. Please check again";
+
+            return this.requestResponse
+        }
+        this.requestResponse.status = "Success";
+        this.requestResponse.statusCode = 200;
+        return this.requestResponse
+
+    }
+
+    /**
+     * Resposible for authenticating user request for system access. Email is 
+     * first verified for existence, then the associated user password hash is 
+     * verified against the existence password hash in the database. Once a match 
+     * is established, the service will request the parameters required to build the
+     * JWT access token. After retrieval, a JWT token is built and sent back to the user
+     * @param userInformation - This params contains only UserName & PassWord
+     */
+    async checkLoginCredentials(userInformation: ILoginInformation): Promise<IResponse> {
+
+        let verifiedEmail: boolean;
+        let savedPasswordHash: string;
+        let accessToken: string;
+
+        try {
+            verifiedEmail = await AuthenticationModel.verifyIfEmailExists(userInformation.email);
+            if (!verifiedEmail)
+                throw new Error();
+        }
+        catch (error) {
+            this.requestResponse.status = "Failure";
+            this.requestResponse.statusCode = 400;
+            this.requestResponse.response = "Email is not in the System or its mispelled. Check Again";
```

Comment:
with the lib i mentionaed above we could just throw new FailureException("Email is not in the System or its mispelled. Check Again"), which makes it one line instead of 3

Reply:
Exceptions & error handling will be updated